WordPress/Classes/ViewRelated/Post/AbstractPostListViewController.swift

```diff
@@ -0,0 +1,1054 @@
+    }
+    
+    func configureNoResultsView() {
+        guard isViewLoaded() == true else {
+            return
+        }
+        
+        let noResultsView = getNoResultsView()
+        
+        if tableViewHandler.resultsController.fetchedObjects?.count > 0 {
+            noResultsView.removeFromSuperview()
+            postListFooterView.hidden = false
+            return
+        }
+        postListFooterView.hidden = true
+        
+        // Refresh the NoResultsView Properties
+        noResultsView.titleText = noResultsTitleText()
+        noResultsView.messageText = noResultsMessageText()
+        noResultsView.accessoryView = noResultsAccessoryView()
+        noResultsView.buttonTitle = noResultsButtonText()
+        
+        // Only add and animate no results view if it isn't already
+        // in the table view
+        if noResultsView.isDescendantOfView(tableView) == false {
+            tableView.addSubviewWithFadeAnimation(noResultsView)
+        } else {
+            noResultsView.centerInSuperview()
+        }
+        
+        tableView.sendSubviewToBack(noResultsView)
+    }
+    
+    func noResultsTitleText() -> String {
+        assert(false, "You should implement this method in the subclass")
+    }
+    
+    func noResultsMessageText() -> String {
+        assert(false, "You should implement this method in the subclass")
+    }
+    
+    func noResultsAccessoryView() -> UIView {
+        if syncHelper.isSyncing {
+            if animatedBox == nil {
+                animatedBox = WPAnimatedBox.newAnimatedBox()
+            }
+            
+            return animatedBox!
+        }
+        
+        return UIImageView(image: UIImage(named: "illustration-posts"))
+    }
+    
+    func noResultsButtonText() -> String? {
+        assert(false, "You should implement this method in the subclass")
+    }
+    
+    func configureAuthorFilter() {
+        assert(false, "You should implement this method in the subclass")
+    }
+    
+    func configureSearchController() {
+        searchController = WPSearchController(searchResultsController: nil)
+        
+        let searchControllerConfigurator = WPSearchControllerConfigurator(searchController: searchController, withSearchWrapperView: searchWrapperView)
+        searchControllerConfigurator.configureSearchControllerAndWrapperView()
+        configureSearchBarPlaceholder()
+        searchController.delegate = self
+        searchController.searchResultsUpdater = self
+    }
+    
+    func configureSearchBarPlaceholder() {
+        // Adjust color depending on where the search bar is being presented.
+        let placeholderColor = WPStyleGuide.wordPressBlue()
+        let placeholderText = NSLocalizedString("Search", comment: "Placeholder text for the search bar on the post screen.")
+        let attrPlacholderText = NSAttributedString(string: placeholderText, attributes: WPStyleGuide.defaultSearchBarTextAttributes(placeholderColor) as? [String:AnyObject])
```

Comment:
Nipticky: for readability's sake, can we move the `attributes`'s `WPStyleGuide` snippet to a `let` constant one line above?.

(Sorry!)